### PR TITLE
feat: generate request telemetry id

### LINKED ISSUE #248: Generate request telemetry ID for Application Insights telemetry model
**Is your feature request related to a problem? Please describe.**
Currently, we use the operation ID as request telemetry ID of the Application Insights model. We should generate this, instead.
Preferable, this generation should be configurable.

**Describe the solution you'd like**
Generate the ID of the request telemetry and provide and option to pass along your own generation function (similar as to the HTTP correlation).
Use this option in the request telemetry converter to have a fresh ID.

**Additional context**
https://github.com/arcus-azure/arcus-service-to-service-correlation-poc

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds an configurable option to generate the telemetry request ID when tracking requests to Azure Application Insights.

Closes #248

### CODE REVIEW

File: docs/preview/02-Features/sinks/azure-application-insights.md [hunk 1]
```diff
@@ -45,6 +45,25 @@ ILogger logger = new LoggerConfiguration()
 
 The Azure Application Insights sink has some additional configuration which can be changed to influence the tracking.
 
+### Requests
+
+### Request ID
+
+When tracking requests, the ID for the request telemetry is by default a generated GUID. The generation of this ID can be configurable via the options.
```
Comment (fgheysels) [inline docs/preview/02-Features/sinks/azure-application-insights.md:52]:
```suggestion
When tracking requests, the ID for the request telemetry is by default a generated GUID. The generation of this ID can be configured via the options.
```

File: docs/preview/02-Features/sinks/azure-application-insights.md [hunk 2]
```diff
@@ -45,6 +45,25 @@ ILogger logger = new LoggerConfiguration()
...
+This is useful (for example) in a service-to-service correlation system where you want the ID of the incoming request based on the sending system, or you want to incorporate the operation ID in the request ID.
```
Comment (fgheysels) [inline docs/preview/02-Features/sinks/azure-application-insights.md:53]:
```suggestion
This is useful (for example) in a service-to-service correlation system where you want the ID of the incoming request to be based on the sending system, or you want to incorporate the operation ID in the request ID.
```

File: src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Configuration/ApplicationInsightsSinkRequestOptions.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    public class ApplicationInsightsSinkRequestOptions
+    {
+        private Func<string> _generatedId = Guid.NewGuid().ToString;
+
+        /// <summary>
+        /// <para>Gets or sets the function to generate the request ID of the telemetry model while tracking requests.</para>
+        /// <para>Default: GUID generation.</para>
+        /// </summary>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="value"/> is <c>null</c>.</exception>
+        public Func<string> GenerateId
```
Comment (fgheysels) [inline src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Configuration/ApplicationInsightsSinkRequestOptions.cs:18]: Would a name like `IdGenerator` be more appropriate for this property ?  Maybe it is more readable when configuring:

```
options => {
    options.IdGenerator = () => .....
}
```

**Review by fgheysels — APPROVED** [review]
Just some small remarks and suggestions.

File: src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Configuration/ApplicationInsightsSinkRequestOptions.cs (hunk 3 above)
Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Configuration/ApplicationInsightsSinkRequestOptions.cs:18]: Hmm, I like to use verbs for functions and methods. We already use `GenerateId` for the correlation info so to be consistent, it seemed appropriate to keep this naming.

**Pull request merged**